### PR TITLE
Add documentation for AEGIS model

### PR DESCRIPTION
**Pull request by ryantwolf**
## Description

Adds documentation for the AEGIS safety classifier.
## Usage

N/A
## Checklist

- [x] I am familiar with the [Contributing Guide](https://github.com/NVIDIA/NeMo-Curator/blob/main/CONTRIBUTING.md).
- [x] New or Existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by jgerh** [comment]
Completed technical edit of file and provided copyedits and suggested revisions for clarity and conciseness.

**Review by jgerh — COMMENTED** [review]
Finished the review

**Review by sarahyurick — COMMENTED** [review]
LGTM, added a few minor comments for your consideration.

File: docs/user-guide/distributeddataclassification.rst [hunk 1]
```diff
@@ -40,47 +38,91 @@ Let's see how ``DomainClassifier`` works in a small excerpt taken from ``example
...
+AEGIS Safety Model
+#####################
+Aegis is a family of content-safety LLMs used for detecting if a piece of text contains content that is a part of 13 critical risk categories.
+There are two variants, `defensive <https://huggingface.co/nvidia/Aegis-AI-Content-Safety-LlamaGuard-Defensive-1.0>`_ and `permissive <https://huggingface.co/nvidia/Aegis-AI-Content-Safety-LlamaGuard-Permissive-1.0>`_, that are useful for filtering harmful data out of your training set.
+The models are parameter-efficient instruction-tuned versions of Llama Guard based on Llama2-7B trained on the NVIDIA content-safety dataset `Aegis Content Safety Dataset <https://huggingface.co/datasets/nvidia/Aegis-AI-Content-Safety-Dataset-1.0>`_.
+More details on training and the model can be found `here <https://arxiv.org/abs/2404.05993>`_.
+
+To use this AEGIS classifiers, you must get access to
+Llama Guard on Hugging Face here: https://huggingface.co/meta-llama/LlamaGuard-7b
+Afterwards, you should set up a user access token and pass that token into
```
[the remark this answers is not in the document]
Comment (ryantwolf) [reply docs/user-guide/distributeddataclassification.rst:66]: I chose a different link, but I agree.

**Comment by ryantwolf** [comment]
Thanks @sarahyurick, let me know what you think now.

**Review by sarahyurick — APPROVED** [review]
LGTM, thanks!

**Pull request merged**